codegens/golang/lib/index.js

```diff
@@ -168,7 +168,10 @@ module.exports = {
         }
 
         if (timeout > 0) {
-            codeSnippet += `${indent}timeout := time.Duration(${timeout} * time.Second)\n`;
+            codeSnippet += `${indent}timeout := time.Duration(${timeout / 1000} * time.Second)\n`;
+        }
+        else {
+            codeSnippet += `${indent}timeout := time.Duration(0 * time.Second)\n`;
```

Comment:
Does 0 mean indefinite?
What happens if we don't add this line at all?

Reply:
A Timeout of zero means no timeout.
The default is 0. They haven't provided anywhere in the official docs but read it on a github issue specifying to use different defaults. Will remove the else block.